### PR TITLE
consensus: fix warnings

### LINKED ISSUE #1271: consensus: fix warnings

### PR DESCRIPTION
**Pull request by bingyanglin**
fix consensus warning
fixes #1271

### CODE REVIEW

**Comment by thibault-martinez** [comment]
@bingyanglin You have conflict because you are pushing 2 commits recently merged in your previous PR

File: consensus/snapshot/snapshots_service.c [hunk 1]
```diff
@@ -288,6 +288,7 @@ static retcode_t iota_snapshots_service_add_entry_point_if_not_orphan(
   retcode_t ret;
   check_not_orphan_do_func_params_t params;
 
+  UNUSED(target_milestone_index);
```
Comment (thibault-martinez) [inline consensus/snapshot/snapshots_service.c:291]: @tsvisabo Why is this parameter nos used ? Can we remove it ?

**Comment by bingyanglin** [comment]
> @bingyanglin You have conflict because you are pushing 2 commits recently merged in your previous PR

File: gossip/components/processor.c [hunk 2]
```diff
@@ -43,7 +43,8 @@ static retcode_t process_transaction_bytes(processor_t const *const processor, t
                                            flex_trit_t const *const hash) {
   retcode_t ret = RC_OK;
   bool exists = false;
-  iota_transaction_t transaction = {.metadata = {.snapshot_index = 0, .solid = 0}, .loaded_columns_mask = {0}};
+  iota_transaction_t transaction;
+  memset(&transaction, 0, sizeof(transaction));
```
Comment (thibault-martinez) [inline gossip/components/processor.c]: Can you call `transaction_reset` on the transaction instead ? And do it after the variables declaration, maybe right after the `if` that checks the parameters

File: consensus/snapshot/snapshots_service.c [hunk 3]
```diff
@@ -152,8 +152,8 @@ retcode_t iota_snapshots_service_generate_snapshot(snapshots_service_t *const sn
   if (ret) {
     return ret;
   }
-  ERR_BIND_RETURN(iota_milestone_service_replay_milestones(snapshots_service->milestone_service, tangle, snapshot,
-                                                           target_milestone->index),
+  ERR_BIND_RETURN(iota_milestone_service_replay_milestones((milestone_service_t *)snapshots_service->milestone_service,
+                                                           (tangle_t *)tangle, snapshot, target_milestone->index),
```
Comment (thibault-martinez) [inline consensus/snapshot/snapshots_service.c]: I would just remove the const from the prototype

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: consensus/snapshot/snapshots_service.c (hunk 3 above)
Comment (bingyanglin) [reply consensus/snapshot/snapshots_service.c]: Do we need the const in the prototype or can just remove it w/o concerns?

Comment (thibault-martinez) [reply consensus/snapshot/snapshots_service.c]: Actually no, I would just add `const` to `milestone_service_t` and `tangle` in `iota_milestone_service_replay_milestones`

File: gossip/components/processor.c (hunk 2 above)
Comment (bingyanglin) [reply gossip/components/processor.c]: Yes, modified. Thank you for pointing this out.

File: consensus/snapshot/snapshots_service.c (hunk 3 above)
Comment (bingyanglin) [reply consensus/snapshot/snapshots_service.c]: Got it, modified. Thanks.

File: consensus/tip_selection/exit_probability_randomizer/tests/test_exit_probability_randomizer.c [hunk 4]
```diff
@@ -224,8 +224,8 @@ void test_single_tx_tangle_walker(void) {
 
 void test_single_tx_tangle_map(void) {
   ep_prob_map_randomizer_t ep_randomizer;
```
Comment (thibault-martinez) [inline consensus/tip_selection/exit_probability_randomizer/tests/test_exit_probability_randomizer.c:226]: This one doesn't need to be initialised 🤔 ?

Comment (bingyanglin) [reply consensus/tip_selection/exit_probability_randomizer/tests/test_exit_probability_randomizer.c:226]: seems the test_cw_gen_topology() takes const variable, and the test_single_tx_tangle_base() takes non-const variable, and const. values should be initialized while they are declared.

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**